packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/handlers/EditLabelEventHandler.java

```diff
@@ -110,10 +110,16 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
             } else {
                 var edge = this.diagramQueryService.findEdgeByLabelId(diagram, input.getLabelId());
                 if (edge.isPresent()) {
-                    this.invokeDirectEditTool(edge.get(), editingContext, diagram, input.getNewText());
-
                     payload = new EditLabelSuccessPayload(diagramInput.getId(), diagram);
                     changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, diagramInput.getRepresentationId(), diagramInput);
+                    try {
+                        this.invokeDirectEditTool(edge.get(), editingContext, diagram, input.getNewText());
+                    } catch (IllegalArgumentException e) {
+                        payload = new ErrorPayload(diagramInput.getId(), this.messageService.invalidNewValue(input.getNewText()));
+                        // We keep the semantic change description to for the diagram to refresh to return to a correct
```

Comment:
I'll change the `changeDescription` object and remove this comment since the proper solution is to fixed https://github.com/eclipse-sirius/sirius-components/issues/301 on the frontend

Reply:
Done